thrift/src/test/java/com/linecorp/armeria/it/metric/PrometheusMetricsIntegrationTest.java

```diff
@@ -16,76 +16,92 @@
+import com.linecorp.armeria.common.metric.MeterIdFunction;
 import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.metric.MetricCollectingService;
 import com.linecorp.armeria.server.metric.PrometheusExporterHttpService;
-import com.linecorp.armeria.server.metric.PrometheusMetricCollectingService;
 import com.linecorp.armeria.server.thrift.THttpService;
 import com.linecorp.armeria.service.test.thrift.main.HelloService.Iface;
 import com.linecorp.armeria.testing.server.ServerRule;
 
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.prometheus.PrometheusMeterRegistry;
+import io.micrometer.core.instrument.util.MeterId;
 import io.prometheus.client.CollectorRegistry;
 
 public class PrometheusMetricsIntegrationTest {
 
-    private static final CollectorRegistry registry = new CollectorRegistry();
+    private static final PrometheusMeterRegistry registry = new PrometheusMeterRegistry();
+    private static final CollectorRegistry prometheusRegistry = registry.getPrometheusRegistry();
 
     @ClassRule
     public static final ServerRule server = new ServerRule() {
         @Override
         protected void configure(ServerBuilder sb) throws Exception {
+            sb.meterRegistry(registry);
+
             final THttpService helloService = THttpService.of((Iface) name -> {
                 if ("world".equals(name)) {
                     return "success";
                 }
                 throw new IllegalArgumentException("bad argument");
             });
 
-            sb.service("/thrift1", helloService.decorate(
-                    PrometheusMetricCollectingService
-                            .newDecorator(registry, MyMetricLabel.values(),
-                                          log -> defaultMetricName(log, "HelloService1"))));
+            sb.service("/foo", helloService.decorate(
+                    MetricCollectingService.newDecorator(
+                            (registry, log) -> meterId(registry, log, "server", "Foo"))));
 
-            sb.service("/thrift2", helloService.decorate(
-                    PrometheusMetricCollectingService
-                            .newDecorator(registry, EnumSet.allOf(MyMetricLabel.class),
-                                          log -> defaultMetricName(log, "HelloService2"))));
+            sb.service("/bar", helloService.decorate(
+                    MetricCollectingService.newDecorator(
+                            (registry, log) -> meterId(registry, log, "server", "Bar"))));
 
             sb.service("/internal/prometheus/metrics",
-                       new PrometheusExporterHttpService(registry));
+                       new PrometheusExporterHttpService(prometheusRegistry));
         }
     };
 
+    private static MeterRegistry oldRegistry;
+
+    @BeforeClass
+    public static void setDefaultMeterRegistry() {
+        oldRegistry = ClientFactory.DEFAULT.meterRegistry();
```

Comment:
It's possible in the future overwriting static fields in tests can be problematic if for example running tests in parallel.  If possible, I guess we shouldn't use the DEFAULT clientfactory in these tests